### ISSUE TITLE
WCAG 2.2: Summary list ‖ Focus Not Obscured (Enhanced)

### ISSUE THREAD

**Issue by querkmachine** [alphagov/govuk-frontend#3841]
## What

In what might be the most pedantic failure of the entire audit: When a summary list row contains enough action links that they wrap onto multiple lines, the dividing line between links can slightly overlap the focus styles of the links on the line above. 

![Zoomed-in screenshot of a focused summary list action link. The grey dividing line from the row below is slightly overlapping the bottom of the summary action link.](https://github.com/alphagov/govuk-frontend/assets/1253214/e25f6dca-9751-4c6b-bff5-f6a3a2ed72d7)

### Resources

- [Summary list component](https://design-system.service.gov.uk/components/summary-list/)
- [WCAG 2.2 criterion for 'Focus Not Obscured (Enhanced)'](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced.html)

## Why

The new WCAG 2.2 Level AAA criterion 'Focus Not Obscured (Enhanced)' says: 

> When a [user interface component](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced.html#dfn-user-interface-component) receives keyboard focus, no part of the component is hidden by author-created content.

This slight overlap is a very technical failure of 'Focus Not Obscured (Enhanced)', which does not permit **any** part of the focus style to be hidden.

It seems like a easy fix, but if it's not, this is probably minor enough to safely ignore. 

## Who needs to work on this
- Developers

## Who needs to review this
- Developers
- Designers (potentially)

## Done when
- [x] The summary list has been amended so that the overlap does not take place
